File: calculator.ts
interface Calculator {
  add: (a: number, b: number) => number;
  subtract: (a: number, b: number) => number;
  multiply: (a: number, b: number) => number;
  divide: (a: number, b: number) => number;
}

class SimpleCalculator implements Calculator {
  add(a: number, b: number) {
    return a + b;
  }
  
  subtract(a: number, b: number) {
    return a - b;
  }
  
  multiply(a: number, b: number) {
    return a * b;
  }
  
  divide(a: number, b: number) {
    if (b === 0) {
      throw new Error('Cannot divide by zero!');
    }
    
    return a / b;
  }
}

// Usage example
const calculator = new SimpleCalculator();
console.log(calculator.add(1, 2)); // Output: 3
console.log(calculator.subtract(5, 3)); // Output: 2
console.log(calculator.multiply(4, 5)); // Output: 20
console.log(calculator.divide(10, 2)); // Output: 5
console.log(calculator.divide(10, 0)); // Throws an error: Cannot divide by zero!

File: calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double num1, num2, result;
        char operator;

        System.out.print("Enter the first number: ");
        num1 = scanner.nextDouble();

        System.out.print("Enter the second number: ");
        num2 = scanner.nextDouble();

        System.out.print("Enter the operator (+, -, *, /): ");
        operator = scanner.next().charAt(0);

        switch (operator) {
            case '+':
                result = num1 + num2;
                break;
            case '-':
                result = num1 - num2;
                break;
            case '*':
                result = num1 * num2;
                break;
            case '/':
                if (num2 == 0) {
                    System.out.println("Error: division by zero");
                    return;
                } else {
                    result = num1 / num2;
                }
                break;
            default:
                System.out.println("Error: invalid operator");
                return;
        }

        System.out.println("Result: " + result);
    }
}
